src/jsx-global.d.ts

```diff
@@ -0,0 +1,10 @@
+import {VNode, VNodeData} from './vnode';
+
+declare global {
+  namespace JSX {
+    type Element = VNode;
+    interface IntrinsicElements {
+      [elemName: string]: VNodeData;
+    }
+  }
+}
```

Comment:
Good question. I don’t think i’ve encountered two jsx providers in one project since the jsxFactory option is set per Typescript project. 

I’m just following the Typescript and react/preact conventions.